backend/apps/github/common.py

```diff
@@ -227,3 +228,96 @@ def sync_repository(
     )
 
     return organization, repository
+
+
+def sync_issue_comments(gh_app, issue: Issue):
+    """Sync new comments for a mentorship program specific issue on-demand.
+
+    Args:
+        gh_app (Github): An authenticated PyGithub instance.
+        issue (Issue): The local database Issue object to sync comments for.
+
+    """
+    logger.info("Starting comment sync for issue #%s", issue.number)
+
+    try:
+        repo = issue.repository
+        if not repo:
+            logger.warning("Issue #%s has no repository, skipping", issue.number)
+            return
+
+        if not repo.owner:
+            logger.warning("Repository for issue #%s has no owner, skipping", issue.number)
+            return
+
+        repo_full_name = f"{repo.owner.login}/{repo.name}"
+        logger.info("Fetching repository: %s", repo_full_name)
+
+        gh_repo = gh_app.get_repo(repo_full_name)
+        gh_issue = gh_repo.get_issue(number=issue.number)
+
+        last_comment = issue.comments.order_by("-created_at").first()
```

Comment:
Add this as a property for Issue -- we have `latest_release`, `latest_pull_request` as examples.